icu4c/source/i18n/unitconverter.cpp

```diff
@@ -56,6 +179,233 @@ MeasureUnit extractCompoundBaseUnit(const MeasureUnit &source, const ConversionR
+ */
+Factor extractFactorConversions(StringPiece stringFactor, UErrorCode &status) {
+    Factor result;
+    SigNum sigNum = SigNum::POSITIVE;
+    auto factorData = stringFactor.data();
+    for (int32_t i = 0, start = 0, n = stringFactor.length(); i < n; i++) {
+        if (factorData[i] == '*' || factorData[i] == '/') {
+            StringPiece factorElement = stringFactor.substr(start, i - start);
+            addFactorElement(result, factorElement, sigNum);
+
+            start = i + 1; // Set `start` to point to the start of the new element.
+        } else if (i == n - 1) {
+            // Last element
+            addFactorElement(result, stringFactor.substr(start, i + 1), sigNum);
+        }
+
+        if (factorData[i] == '/')
+            sigNum = SigNum::NEGATIVE; // Change the sigNum because we reached the Denominator.
+    }
+
+    return result;
+}
+
+// Load factor for a single source
+Factor loadSingleFactor(StringPiece source, const ConversionRates &ratesInfo, UErrorCode &status) {
+    const auto conversionUnit = ratesInfo.extractConversionInfo(source, status);
+    if (U_FAILURE(status)) return Factor();
+    if(conversionUnit == nullptr) {
+        status = U_INTERNAL_PROGRAM_ERROR;
+        return Factor();
+    }
+
+    Factor result = extractFactorConversions(conversionUnit->factor.toStringPiece(), status);
+    result.offset = strHasDivideSignToDouble(conversionUnit->offset.toStringPiece());
+
+    return result;
+}
+
+// Load Factor of a compound source unit.
+Factor loadCompoundFactor(const MeasureUnit &source, const ConversionRates &ratesInfo,
+                          UErrorCode &status) {
+
+    Factor result;
+    auto compoundSourceUnit = MeasureUnitImpl::forMeasureUnitMaybeCopy(source, status);
+    if (U_FAILURE(status)) return result;
+
+    for (int32_t i = 0, n = compoundSourceUnit.units.length(); i < n; i++) {
+        auto singleUnit = *compoundSourceUnit.units[i]; // a TempSingleUnit
+
+        Factor singleFactor = loadSingleFactor(singleUnit.identifier, ratesInfo, status);
+        if(U_FAILURE(status)) return result;
+
+        // Apply SiPrefix before the power, because the power may be will flip the factor.
+        singleFactor.applySiPrefix(singleUnit.siPrefix);
+
+        // Apply the power of the `dimensionality`
+        singleFactor.power(singleUnit.dimensionality);
+
+        result.multiplyBy(singleFactor);
+    }
+
+    return result;
+}
+
+void substituteSingleConstant(int32_t constantPower,
+                              double constantValue /* constant actual value, e.g. G= 9.88888 */,
+                              Factor &factor) {
+    constantValue = std::pow(constantValue, std::abs(constantPower));
+
+    if (constantPower < 0) {
+        factor.factorDen *= constantValue;
+    } else {
+        factor.factorNum *= constantValue;
+    }
+}
+
+void substituteConstants(Factor &factor, UErrorCode &status) {
```

Comment:
Suggestion: make this a method on `Factor`.

Reply:
For example, have `Factor::substituteConstants`, such that you can call `factor.substituteConstants(status)`.

P.S. Please reply to the thread; don't edit my comment with your answer.  It's confusing.